### PR TITLE
154

### PR DESCRIPTION
**Pull request by ccemeraldeyes**
#154 

refactor/simplify broker config.

### CODE REVIEW

File: brokerapi/brokers/config/broker_config.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+	"os"
+)
+
+type BrokerConfig struct {
+	Catalog    map[string]models.Service
+	HttpConfig *jwt.Config
+	ProjectId  string
+}
+
+func NewBrokerConfig() (*BrokerConfig, error) {
```
Comment (johnsonj) [inline brokerapi/brokers/config/broker_config.go]: suggest: NewBrokerConfigFromEnv

File: brokerapi/brokers/stackdriver_trace/broker.go [hunk 2]
```diff
@@ -56,7 +48,7 @@ func (b *StackdriverTraceBroker) Bind(instanceID, bindingID string, details mode
 	details.Parameters["role"] = "cloudtrace.agent"
 
 	// Create account
-	newBinding, err := b.ServiceAccountManager.CreateAccountInGoogle(instanceID, bindingID, details, models.ServiceInstanceDetails{})
+	newBinding, err := b.AccountManager.CreateAccountInGoogle(instanceID, bindingID, details, models.ServiceInstanceDetails{})
```
Comment (johnsonj) [inline brokerapi/brokers/stackdriver_trace/broker.go]: now that it's AccountManager there's nothing to indicate ServiceAccount. Google account is pretty generic. Suggest renaming method to something like CreateServiceAccount

Comment (ccemeraldeyes) [inline brokerapi/brokers/stackdriver_trace/broker.go]: Why do you want it to indicate service account?

Comment (johnsonj) [inline brokerapi/brokers/stackdriver_trace/broker.go]: So it indicates what it's actually making. AccountInGoogle doesn't have a ton of meaning. What kind of account? google what? a gmail?

File: brokerapi/brokers/config/broker_config.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	"fmt"
+	"gcp-service-broker/brokerapi/brokers/models"
+	"gcp-service-broker/utils"
+	"golang.org/x/oauth2/jwt"
+	"gopkg.in/validator.v2"
+	"os"
+)
+
+type BrokerConfig struct {
+	Catalog    map[string]models.Service
```
Comment (mkjelland) [inline brokerapi/brokers/config/broker_config.go]: Do you need to store this as a map here? If not, you could store it as a list and then you won't need a public ServiceList method.

File: brokerapi/brokers/stackdriver_trace/broker.go (hunk 2 above)
Comment (ccemeraldeyes) [inline brokerapi/brokers/stackdriver_trace/broker.go]: I guess my point is that these brokers are meant to be pretty generic. It shouldn't really matter what kind of account. And the kind of account *is* indicated on creation of the broker. And to have a consistent account manager interface, a generic name is needed anyway.

Comment (johnsonj) [inline brokerapi/brokers/stackdriver_trace/broker.go]: well they're generic in that they are provisioning a multitude of services but aren't they always GCP services and GCP service accounts? is that ever going to be different?

File: brokerapi/brokers/config/broker_config.go [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+func NewBrokerConfigFromEnv() (*BrokerConfig, error) {
+	var err error
+	bc := BrokerConfig{}
+	creds, err := bc.GetCredentialsFromEnv()
+	if err != nil {
+		return &BrokerConfig{}, err
+	}
+	bc.ProjectId = creds.ProjectId
+	conf, err := utils.GetAuthedConfig()
```
Comment (mkjelland) [inline brokerapi/brokers/config/broker_config.go]: Can GetAuthedConfig be moved out of utils and into this file? This file has a more specific purpose than utils and contains other similar methods.

File: brokerapi/brokers/stackdriver_trace/broker.go (hunk 2 above)
Comment (ccemeraldeyes) [inline brokerapi/brokers/stackdriver_trace/broker.go]: CloudSQL ruins everything, yes.

Comment (johnsonj) [inline brokerapi/brokers/stackdriver_trace/broker.go]: ok, what about CreateAccount/Create/CreateCredentials?

'InGoogle' doesn't sound like it's creating a sql username/password

Comment (ccemeraldeyes) [inline brokerapi/brokers/stackdriver_trace/broker.go]: sold.

Comment (johnsonj) [inline brokerapi/brokers/stackdriver_trace/broker.go]: thank you and sorry for the super nit of method names past :)

File: brokerapi/brokers/config/broker_config.go (hunk 4 above)
Comment (ccemeraldeyes) [inline brokerapi/brokers/config/broker_config.go]: it's used in the db migrations code too so I'm going to leave it.

File: brokerapi/brokers/config/broker_config.go (hunk 3 above)
Comment (ccemeraldeyes) [inline brokerapi/brokers/config/broker_config.go]: I'm now also storing it as a map on the broker, which is what I meant to do ;)

**Pull request merged**